test/unit/algorithm_mode/test_serve_utils.py

```diff
@@ -76,3 +82,180 @@ def test_predict_valid_content_type(correct_content_type):
     mock_dmatrix.feature_names = mock_feature_names
 
     serve_utils.predict(mock_booster, serve_utils.PKL_FORMAT, mock_dmatrix, correct_content_type)
+
+
+TEST_RAW_PREDICTIONS = np.array([0.6, 0.1])
+TEST_KEYS = [serve_utils.PREDICTED_LABEL, serve_utils.PROBABILITIES]
+TEST_PREDICTIONS = [
+    {serve_utils.PREDICTED_LABEL: 1, serve_utils.PROBABILITIES: [0.4, 0.6]},
+    {serve_utils.PREDICTED_LABEL: 0, serve_utils.PROBABILITIES: [0.9, 0.1]}
+]
+
+TEST_KEYS_BINARY_LOG = serve_utils.VALID_OBJECTIVES[serve_utils.BINARY_LOG]
+TEST_PREDICTIONS_BINARY_LOG = [
+    {serve_utils.PREDICTED_LABEL: 1, serve_utils.LABELS: [0, 1], serve_utils.PROBABILITY: 0.6,
+     serve_utils.PROBABILITIES: [0.4, 0.6], serve_utils.RAW_SCORE: 0.6, serve_utils.RAW_SCORES: [0.4, 0.6]},
+    {serve_utils.PREDICTED_LABEL: 0, serve_utils.LABELS: [0, 1], serve_utils.PROBABILITY: 0.1,
+     serve_utils.PROBABILITIES: [0.9, 0.1], serve_utils.RAW_SCORE: 0.1, serve_utils.RAW_SCORES: [0.9, 0.1]}
+]
+
+TEST_RAW_PREDICTIONS_REG_LOG = np.array([0.5, -7.0])
+TEST_KEYS_REG_LOG = serve_utils.VALID_OBJECTIVES[serve_utils.REG_LOG]
+TEST_PREDICTIONS_REG_LOG = [{"predicted_score": 0.5}, {"predicted_score": -7.0}]
+
+
+def test_is_selectable_inference_response_false():
+    assert not serve_utils.is_selectable_inference_output()
+
+
+def test_is_selectable_inference_response_true():
+    os.environ['SAGEMAKER_INFERENCE_OUTPUT'] = 'predicted_label'
+    assert serve_utils.is_selectable_inference_output()
+    del os.environ['SAGEMAKER_INFERENCE_OUTPUT']
+
+
+def test_get_selected_content_keys():
+    os.environ['SAGEMAKER_INFERENCE_OUTPUT'] = 'predicted_label'
```

Comment:
Instead of changing this value in the environment, you should be using a monkey patch or similar for this test.